### ISSUE TITLE
Improve component preview in the side panel

### ISSUE THREAD

**Issue by johno** [blocks/blocks#62]
We may need to place the rendered component into an iframe and then scale it.

**Comment by souporserious** [comment]
I was thinking about this when doing viewports. Thoughts on an equally sized box with a small preview similar to FramerX? I love that it's actual code in the sidebar, but maybe using generated screenshots might be better for performance?

**Comment by johno** [comment]
Yeah, the tricky part with generated screenshots is we'd have to send along the entire scope and code on the fly somehow (since we're on the client), so if we could figure out a way to make the code work that'd be super rad.

**Comment by souporserious** [comment]
Ah yeah, that adds way more complexity. If perf becomes an issue, windowing might be a nice way to offload loading too many components. I'll play around with this after viewports if no-one gets to it before then.

**Comment by jordanoverbye** [comment]
I've also noticed that if you change the the Theme to something with a dark background, the component previews don't open. I think *some*  things in the component preview update (like text color and font family) but not all

**Comment by jordanoverbye** [comment]
@johno What is the expected behaviour? Should the component preview inherit all of the properties of the active theme?

**Comment by johno** [comment]
Yeah, I'd expect the component preview to inherit all properties from the active theme so that it's representative of what will be rendered on the canvas if drag and dropped.

**Comment by vladshcherbin** [comment]
As per scaling, you can try using css [zoom property](https://developer.mozilla.org/en-US/docs/Web/CSS/zoom). Here is how it looks with 0.5:

![image](https://user-images.githubusercontent.com/6711845/85201840-0817a180-b30b-11ea-8d15-9c6b8b90004c.png)


Saw it [here](https://theme-ui.com/recipes).
